src/peer.js

```diff
@@ -87,6 +87,8 @@ class Peer extends EventEmitter {
     this.conn = conn
     this.stream = pushable({
       onEnd: () => {
+        // close readable side of the stream
+        this.conn.reset && this.conn.reset()
```

Comment:
As a follow-up, I will create an issue to remove this validation and create a wrapper around `it-pair` for these tests.

[it-pair](https://github.com/alanshaw/it-pair) provides a bare minimal duplex stream for testing, but it does not have the reset function, as a multiplex stream. I tested this with `js-libp2p` and the stream reset function is called there.